pkg/lang/golang/plugin_expose.go

```diff
@@ -217,21 +224,45 @@ func (h *restAPIHandler) handleFile(f *core.SourceFile) (*core.SourceFile, error
 
 		gwSpec := gatewaySpec{
 			FilePath:   f.Path(),
-			AppVarName: appName,
+			AppVarName: routerName,
 			Id:         cap.ID,
 		}
 
-		log = log.With(zap.String("var", appName))
+		log = log.With(zap.String("var", routerName))
 
-		localRoutes, err := h.findChiRoutesForVar(f, appName, "")
+		localRoutes, err := h.findChiRoutesForVar(f, routerName, "")
 		if err != nil {
 			return nil, core.NewCompilerError(f, capNode, err)
 		}
 
 		if len(localRoutes) > 0 {
-			log.Sugar().Infof("Found %d route(s) on app '%s'", len(localRoutes), appName)
+			log.Sugar().Infof("Found %d route(s) on app '%s'", len(localRoutes), routerName)
 			h.RoutesByGateway[gwSpec] = append(h.RoutesByGateway[gwSpec], localRoutes...)
 		}
+
+		// For external routes, we work back from the mount() call to get the package being called. Then
+		// we find the function which defines the extra routes within the specified package
+
+		routerMounts := h.findChiRouterMounts(f, routerName)
```

Comment:
this is still limited to the same file as the listen call right?

Reply:
when checking for external calls, we no longer care about the listen. So we're just getting the package name of the mounted package and finding the file where the routes are defined. It's no longer in the main.go file w/ the listen anymore. Not sure if that answered your question

Reply:
this is only run when a file has annotations  though and weve found the listener. Im not sure if go allows the routes to be in a separate file, but is that working? Im not sure if your answer is clear to me, im a little confused